backend/TreeOfAKind.Domain/UserProfiles/UserProfile.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Globalization;
+using System.Threading;
+using NodaTime;
+using TreeOfAKind.Domain.SeedWork;
+
+namespace TreeOfAKind.Domain.UserProfiles
+{
+    public class UserProfile : Entity, IAggregateRoot
+    {
+        public UserId Id { get; private set; }
+
+        public string AuthUserId { get; set; }
+
+        public DateTime? BirthDate { get; private set; }
+
+        public string? FirstName { get; private set; }
+        
+        public string? LastName { get; private set; }
+
+        public string? Username { get; private set; }
+
+        public string? Description { get; private set; }
+
+        private UserProfile()
+        {
+            Id = default!;
+            AuthUserId = default!;
+            BirthDate = default!;
+            FirstName = default!;
+            LastName = default!;
+            Username = default!;
+            Description = default!;
+        }
+
+        private UserProfile(string authUserId, DateTime? birthDate, string? firstName, string? lastName, string? username, string? description)
+        {
+            Id = new UserId(new Guid());
+            AuthUserId = authUserId;
+            BirthDate = birthDate;
+            FirstName = lastName;
+            LastName = lastName;
+            Username = username;
+            Description = description;
+            
+            AddDomainEvent(new UserCreatedEvent(Id));
+        }
+
+        public static UserProfile CreateUser(
+            string authUserId,
+            DateTime? birthDate,
+            string firstName,
+            string lastName,
+            string userName,
+            string description,
+            IUsernameUniquenessChecker usernameUniquenessChecker,
```

Comment:
😕

Reply:
That is a business rule and does not need an interface in the API of the domain, since someone could inject here a bad validator, which would allow to create invalid instance.